code/binary-tree/src/test/java/ru/unn/agile/binaryTree/model/BinaryTreeFunctionalTests.java

```diff
@@ -0,0 +1,109 @@
+package ru.unn.agile.binarytree.model;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Random;
+
+import static org.junit.Assert.*;
+
+public class BinaryTreeFunctionalTests {
+    private BinaryTree binaryTree;
+
+    private int     simpleKey   = 1;
+    private String  simpleData  = "FirstSimpleText";
+
+    private int bound = 1000;
+    private int arrSize = 10;
+    private static Random randomGen = new Random();
+
+    @Before
+    public void setUp() {
+        binaryTree = new BinaryTree();
+    }
+
+    @Test
+    public void canAddAfterDeletingRoot() {
+        binaryTree.add(simpleKey, simpleData);
+        final int    secondKey  = 2;
+        final String secondData = "SecondSimpleText";
+
+        assertTrue(binaryTree.remove(simpleKey));
+
+        binaryTree.add(secondKey, secondData);
+    }
+
+    @Test
+    public void canAddTenValuesAndGetIt() {
+        final int size = 10;
+        for (int i = 0; i < size; ++i) {
+            binaryTree.add(i, Integer.toString(i));
+        }
+
+        for (int i = 0; i < size; ++i) {
+            assertEquals(Integer.toString(i), binaryTree.find(i));
+        }
+    }
+
+    @Test
+    public void canAddTenRemoveMiddleAndGetRest() {
+        final int middleKey = 5;
+        final int size      = 10;
+
+        for (int i = 0; i < size; ++i) {
+            binaryTree.add(i, Integer.toString(i));
+        }
+
+        assertTrue(binaryTree.remove(middleKey));
+
+        for (int i = 0; i < size; ++i) {
+            if (i != middleKey) {
+                assertEquals(Integer.toString(i), binaryTree.find(i));
+            } else {
+                assertNull(binaryTree.find(i));
+            }
```

Comment:
It's better to skip this else branch because we don't want to check this. The assertion should be only one to correctly test the statement